python/lsst/pipe/tasks/postprocess.py

```diff
@@ -329,75 +333,359 @@ def run(self, catalog, ccdVisitId=None):
+    def attachCalibs(self, inputRefs, skyMap, exposure, externalSkyWcsGlobalCatalog=None,
+                     externalSkyWcsTractCatalog=None, externalPhotoCalibGlobalCatalog=None,
+                     externalPhotoCalibTractCatalog=None, **kwargs):
+        """Apply external calibrations to exposure per configuration
+
+        When multiple tract-level calibrations overlap, select the one with the
+        center closest to detector.
 
         Parameters
         ----------
-        catalog: `afwTable.SourceCatalog`
-            catalog to which calib columns will be added
-        dataRef: `lsst.daf.persistence.ButlerDataRef
-            for fetching the calibs from disk.
+        inputRefs : `lsst.pipe.base.InputQuantizedConnection`, for dataIds of
+            tract-level calibs.
+        skyMap : `lsst.skymap.SkyMap`
+        exposure : `lsst.afw.image.exposure.Exposure`
+            Input exposure to adjust calibrations.
+        externalSkyWcsGlobalCatalog : `lsst.afw.table.ExposureCatalog`, optional
+            Exposure catalog with external skyWcs to be applied per config
+        externalSkyWcsTractCatalog : `lsst.afw.table.ExposureCatalog`, optional
+            Exposure catalog with external skyWcs to be applied per config
+        externalPhotoCalibGlobalCatalog : `lsst.afw.table.ExposureCatalog`, optional
+            Exposure catalog with external photoCalib to be applied per config
+        externalPhotoCalibTractCatalog : `lsst.afw.table.ExposureCatalog`, optional
+
 
         Returns
         -------
-        newCat:  `afwTable.SourceCatalog`
-            Source Catalog with requested local calib columns
+        exposure : `lsst.afw.image.exposure.Exposure`
+            Exposure with adjusted calibrations.
         """
-        mapper = afwTable.SchemaMapper(catalog.schema)
-        measureConfig = SingleFrameMeasurementTask.ConfigClass()
-        measureConfig.doReplaceWithNoise = False
+        if self.config.useGlobalExternalSkyWcs:
+            externalSkyWcsCatalog = externalSkyWcsGlobalCatalog
+            self.log.info('Applying global SkyWcs')
+        else:
+            # get closest overlapping tract
+            if externalSkyWcsTractCatalog:
+                inputRef = getattr(inputRefs, 'externalSkyWcsTractCatalog')
+                tracts = [ref.dataId['tract'] for ref in inputRef]
+                if len(tracts) == 1:
+                    ind = 0
+                    self.log.info('Applying tract-level SkyWcs from tract %s', tracts[ind])
+                else:
+                    ind = self.getClosestTract(tracts, skyMap,
+                                               exposure.getBBox(), exposure.getWcs())
+                    self.log.info('Multiple overlapping externalSkyWcsTractCatalogs found (%s). '
+                                  'Applying closest to detector center: tract=%s', str(tracts), tracts[ind])
+            externalSkyWcsCatalog = externalSkyWcsTractCatalog[ind]
```

Comment:
Same comment here about whether this line should be inside `if externalSkyWcsTractCatalog`. I think it has to be otherwise you could get a situation where ind is not defined (unless you can be 100% confident that externalSkyWcsTractCatalog will always be there).